nilearn/image/image.py

```diff
@@ -842,7 +862,45 @@ def threshold_img(img, threshold, mask_img=None, copy=True):
                                            percentile_func=scoreatpercentile,
                                            name='threshold')
 
-    img_data[np.abs(img_data) < cutoff_threshold] = 0.
+    # Apply threshold
+    if two_sided:
+        img_data[np.abs(img_data) < cutoff_threshold] = 0.
+    else:
+        img_data[img_data < cutoff_threshold] = 0.
+
+    # Expand to 4D to support both 3D and 4D
+    expand_to_4d = img_data.ndim == 3
+    if expand_to_4d:
+        img_data = img_data[:, :, :, None]
+
+    # Perform cluster thresholding, if requested
+    if cluster_threshold > 0:
+        # Define array for 6-connectivity, aka NN1 or "faces"
+        conn_mat = np.zeros((3, 3, 3), int)
+        conn_mat[1, 1, :] = 1
+        conn_mat[1, :, 1] = 1
+        conn_mat[:, 1, 1] = 1
+
+        for i_vol in range(img_data.shape[3]):
+            vol_data = img_data[..., i_vol]
+            for sign in np.sign(vol_data):
+                # Binarize using one-sided cluster-defining threshold
+                binarized = ((vol_data * sign) > 0).astype(int)
+
+                # Apply cluster threshold
+                label_map = ndimage.measurements.label(binarized, conn_mat)[0]
+                clust_ids = sorted(list(np.unique(label_map)[1:]))
+                for c_val in clust_ids:
+                    if np.sum(label_map == c_val) < cluster_threshold:
+                        vol_data[label_map == c_val] = 0
+
+            img_data[..., i_vol] = vol_data
+
+    if expand_to_4d:
+        # Reduce back to 3D
+        img_data = img_data[:, :, :, 0]
+
+    # Reconstitute img object
     threshold_img = new_img_like(img, img_data, affine)
```

Comment:
Should copy_header be set to the default (False) here? It seems like it should retain the header info, but I'm not sure if there's an issue with `resample_img` above.